dockerfiles/docker-compose.yml

```diff
@@ -0,0 +1,117 @@
+
+# This composition runs a full vdb instance with automatic service restarts,
+# preconfigured for mainnet.
+#
+# When running you can access:
+# postgres   @ localhost:5432
+# graph(i)ql @ localhost:5000
+#
+# The client and database service stores the data on the *host* machine in a docker volume.
+# This means that if the cluster is gracefully stopped, it can be restarted again
+# without data loss.
+#
+# Commands:
+# To start the cluster: `> docker-compose up`
+# To stop the cluster, just `ctrl+c` (once, to allow a graceful exit)
+# To reset the cluster: `docker-compose down --volumes` (after stopping the cluster)
+#
+# Note that it'll take a couple of days for the client to sync with thge network before
+# vdb can do it's work.
+
+version: "3.7"
+services:
+  header-sync:
+    image: makerdao/vdb-headersync:latest # (needs wait-for-it.sh)
+    environment:
+      - CLIENT_IPCPATH=ws://geth-statediffing:8546
+      - DATABASE_NAME=vdb
+      - DATABASE_HOSTNAME=db
+      - DATABASE_PORT=5432
+      - DATABASE_USER=vdb
+      - DATABASE_PASSWORD=vdb
+      - STARTING_BLOCK_NUMBER=8928152
+    depends_on:
+      - db
+    command: ["./wait-for-it.sh", "db:5432", "--strict", "--", "./startup_script.sh"]
+    restart: unless-stopped
+
+  extract-diffs:
+    image: makerdao/vdb-extract-diffs:latest # (needs wait-for-it.sh)
+    environment:
+      # Remote client would need to be statediff enabled
+      - CLIENT_IPCPATH=ws://geth-statediffing:8546
+      - STORAGEDIFFS_SOURCE=geth
+      - DATABASE_NAME=vdb
+      - DATABASE_HOSTNAME=db
+      - DATABASE_PORT=5432
+      - DATABASE_USER=vdb
+      - DATABASE_PASSWORD=vdb
+    depends_on:
+      - db
+    command: ["./wait-for-it.sh", "db:5432", "--strict", "--", "./startup_script.sh"]
+
+  execute-mcd:
+    image: makerdao/vdb-mcd-execute:latest # (needs wait-for-it.sh)
+    environment:
+      - CLIENT_IPCPATH=ws://geth-statediffing:8546
```

Comment:
idk if this matters but I don't think this needs to be websockets for `execute`

Reply:
🤷 Matches the way we did it elsewhere.